frontend/src/components/ConfirmationDialogProvider.tsx

```diff
@@ -0,0 +1,84 @@
+import React, {
+    createContext,
+    useCallback,
+    useContext,
+    useRef,
+    useState,
+} from 'react';
+import { Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle } from '@mui/material'
+import { StyledButton } from './styles';
+// Built from this example buth with MUI dialogs: https://akashhamirwasia.com/blog/building-expressive-confirm-dialog-api-in-react/
+// Uses a context provider to allow any component to access a confirm dialog component using the useConfirm hook
+// Example: 
+// import useConfirm from '../../ConfirmationDialogProvider';
+// const confirm = useConfirm()
+// const confirmed = await confirm(
+//      {
+//          title: 'Confirm This Action', 
+//          message: "Are you sure you want to do this?"
+//      })
+
+interface ConfirmData {
+    title: string
+    message: string
+}
+
+type confirmContext = (data: ConfirmData) => Promise<boolean>
+
+const ConfirmDialog = createContext<confirmContext>(null);
+
+export function ConfirmDialogProvider({ children }) {
+
+
+    const [state, setState] = useState({ isOpen: false, title: '', message: '' });
```

Comment:
nit

One thing we need to be careful of with this pattern is, manipulating state (like state.isOpen = true), won't trigger a refresh. Only reinstantiating the object with trigger a refresh

Looks like that's what's happening here, and I see that it makes the code cleaner, but I wanted to mention it anyway so that we're careful with it in the future

Reply:
Using the setState function in the callback should reinstate the object though right?

Reply:
Yep, it's being reinstantiated in the callback, so it's fine